product_attribute_types/models/product.py

```diff
@@ -0,0 +1,47 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see http://www.gnu.org/licenses/.
+#
+##############################################################################
+
+from openerp import models, fields
+
+
+class ProductAttribute(models.Model):
+    _inherit = "product.attribute"
+
+    attr_type = fields.Selection(required=True, selection=[
+        ('select', 'Select'),
+        ('range', 'Range'),
+        ('numeric', 'Numeric')], string="Type", default='select')
+
+
+class ProductAttributeLine(models.Model):
+    _inherit = "product.attribute.line"
+
+    required = fields.Boolean('Required')
+    default = fields.Many2one('product.attribute.value', 'Default')
+    attr_type = fields.Selection(string='Type', store=False,
+                                 related='attribute_id.attr_type')
+
+
+class ProductAttributeValue(models.Model):
+    _inherit = "product.attribute.value"
+
+    attr_type = fields.Selection(string='Type',
+                                 related='attribute_id.attr_type')
+    numeric_value = fields.Float('Numeric Value', digits=(12, 6))
```

Comment:
The problem is that depending on the use of this field, the precision can be different. For now, we use it for at least three things:

* the quantity of the product associated in another field (module _mrp_bom_through_attributes_).
* the conversion factor between packages (module _product_packaging_through_attributes_).
* another one I didn't remember just now :wink: 

so I let it with enough resolution for each use case.

Anyways, you can create a new decimal precision for this, and let the user decides, but don't put any of the existing ones.